### PR TITLE
Datapack parsing

### PR DESCRIPTION
**Pull request by koskja**
# Datapack parsing

## Status

- [ ] Ready 
- [x] Development
- [ ] Hold

## Description

The game needs to parse registries for most of its logic to function. This pull request attempts to do that.

## Related issues

#429

## Checklist

- [ ] Ran `cargo fmt`, `cargo clippy --all-targets`, `cargo build --release` and `cargo test` and fixed any generated errors!
- [ ] Removed unnecessary commented out code
- [ ] Used specific traces (if you trace actions please specify the cause i.e. the player)

Note: if you locally don't get any errors, but GitHub Actions fails (especially at `clippy`) you might want to check your rust toolchain version. You can then feel free to fix these warnings/errors in your PR.

### CODE REVIEW

**Comment by koskja** [comment]
The functions now take `T: Into<NamespacedId> + Clone`. @caelunshun

File: feather/datapacks/tags.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from os import listdir
```
Comment (Defman) [inline feather/datapacks/tags.py:1]: Should be moved into the generator directory.

**Review by Defman — CHANGES_REQUESTED** [review]
Looks good, the location of the generator.py is a bit odd and should be moved into the same directory with all other generators?

**Comment by koskja** [comment]
Moved the tags generator. I believe the recipe implementation can be finished together with crafting.

**Pull request merged**